# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/html/HTMLImageElement.cpp
# Commit: 2de3021ee59b
# Full Hash: 2de3021ee59b2fd6be0792548ce10ec45c45ca51
# Author: Boris Chiou <boris.chiou@gmail.com>
# Date: 2022-08-23 02:46:57
# Regressor Bug: 1694741
# File Overlap Count: 1
# Description:
#   Bug 1694741 - Part 2: Factor out the check if the parent is <picture> for <source> and <img>. r=emilio
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D152708
# ==============================================================================

diff -r 21ad6900a8ad -r 2de3021ee59b dom/html/HTMLImageElement.cpp
--- a/dom/html/HTMLImageElement.cpp	Mon Aug 22 20:18:37 2022 +0000
+++ b/dom/html/HTMLImageElement.cpp	Mon Aug 22 20:18:37 2022 +0000
@@ -831,8 +831,7 @@
     return true;
   }
 
-  Element* parent = GetParentElement();
-  return parent && parent->IsHTMLElement(nsGkAtoms::picture);
+  return IsInPicture();
 }
 
 bool HTMLImageElement::InResponsiveMode() {
@@ -1030,12 +1029,11 @@
 
   nsIContent* currentSource =
       mResponsiveSelector ? mResponsiveSelector->Content() : nullptr;
-  Element* parent = nsINode::GetParentElement();
 
   nsINode* candidateSource = nullptr;
-  if (parent && parent->IsHTMLElement(nsGkAtoms::picture)) {
+  if (IsInPicture()) {
     // Walk source nodes previous to ourselves
-    candidateSource = parent->GetFirstChild();
+    candidateSource = GetParentElement()->GetFirstChild();
   } else {
     candidateSource = this;
   }
@@ -1111,8 +1109,7 @@
 bool HTMLImageElement::SourceElementMatches(Element* aSourceElement) {
   MOZ_ASSERT(aSourceElement->IsHTMLElement(nsGkAtoms::source));
 
-  DebugOnly<Element*> parent(nsINode::GetParentElement());
-  MOZ_ASSERT(parent && parent->IsHTMLElement(nsGkAtoms::picture));
+  MOZ_ASSERT(IsInPicture());
   MOZ_ASSERT(IsPreviousSibling(aSourceElement, this));
 
   // Check media and type